cmd/fs-v1.go

```diff
@@ -404,13 +404,15 @@ func (fs *FSObjects) ListBuckets(ctx context.Context) ([]BucketInfo, error) {
 
 // DeleteBucket - delete a bucket and all the metadata associated
 // with the bucket including pending multipart, object metadata.
-func (fs *FSObjects) DeleteBucket(ctx context.Context, bucket string) error {
-	bucketLock := fs.NewNSLock(ctx, bucket, "")
-	if err := bucketLock.GetLock(globalObjectTimeout); err != nil {
-		logger.LogIf(ctx, err)
-		return err
+func (fs *FSObjects) DeleteBucket(ctx context.Context, bucket string, forceDelete bool) error {
+	if !forceDelete {
```

Comment:
Not clear why we do not lock when force delete is passed, like a delete in the middle of download ?

Reply:
On purpose, there is no locking for buckets @vadmeste if there is any active upload/download that will crash. This is considered a disastrous operation.